src/Security/Security.php

```diff
@@ -596,180 +677,180 @@ public static function clearLoginMessage()
      * For multiple authenticators, Security_MultiAuthenticatorLogin is used.
      * See getTemplatesFor and getIncludeTemplate for how to override template logic
      *
-     * @return string|HTTPResponse Returns the "login" page as HTML code.
+     * @param $request
+     * @return HTTPResponse|string Returns the "login" page as HTML code.
+     * @throws HTTPResponse_Exception
      */
-    public function login()
+    public function login($request, $service = Authenticator::LOGIN)
```

Comment:
This brings back the `default_authenticator`, otherwise the CMSSecurity has to awkwardly create a request if the request is null, to pass in the cms name.